app/src/main/java/com/hover/stax/views/CustomTextInputLayout.java

```diff
@@ -0,0 +1,120 @@
+package com.hover.stax.views;
+
+import android.content.Context;
+import android.content.res.ColorStateList;
+import android.content.res.TypedArray;
+import android.content.res.XmlResourceParser;
+import android.util.AttributeSet;
+import android.view.LayoutInflater;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.google.android.material.textfield.TextInputEditText;
+import com.google.android.material.textfield.TextInputLayout;
+import com.hover.stax.R;
+import com.hover.stax.utils.Utils;
+
+public class CustomTextInputLayout extends TextInputLayout {
```

Comment:
I created a separate object for dropdown and text; cos of:
1. Different styling.
2. Different View type inside the TextInputLayout. Dropdown uses Autocomplete, while text uses an edit. 

I thought of it being cleaner when separated for anyone reading the code; and dropdown functionality could be extended for whatever reason without having it affect texts.

Reply:
It should work like the base material design TextInputLayout so that you can pass either a EditText or AutoComplete as the child depending on which you need. That is a serious reduction in code duplication and I don't believe there are any style differences between the TextInput part, the child is where the style differences are.

I did actually make an attempt to combine these to see if it is possible, but ran into issues with the children being added in the right place, so if it is too much work to combine then having different classes is fine for now.